pkg/remote/aws/nat_gateway_supplier_test.go

```diff
@@ -28,50 +28,34 @@ func TestNatGatewaySupplier_Resources(t *testing.T) {
 	cases := []struct {
 		test    string
 		dirName string
-		mocks   func(client *mocks.FakeEC2)
+		mocks   func(repo *repository.MockEC2Repository)
 		err     error
 	}{
 		{
 			test:    "no gateway",
 			dirName: "nat_gateway_empty",
-			mocks: func(client *mocks.FakeEC2) {
-				client.On("DescribeNatGatewaysPages",
-					&ec2.DescribeNatGatewaysInput{},
-					mock.MatchedBy(func(callback func(res *ec2.DescribeNatGatewaysOutput, lastPage bool) bool) bool {
-						callback(&ec2.DescribeNatGatewaysOutput{}, true)
-						return true
-					})).Return(nil)
+			mocks: func(repo *repository.MockEC2Repository) {
+				repo.On("ListAllNatGateways").Once().Return([]*ec2.NatGateway{}, nil)
 			},
 			err: nil,
 		},
 		{
 			test:    "single aws_nat_gateway",
 			dirName: "nat_gateway",
-			mocks: func(client *mocks.FakeEC2) {
-				client.On("DescribeNatGatewaysPages",
-					&ec2.DescribeNatGatewaysInput{},
-					mock.MatchedBy(func(callback func(res *ec2.DescribeNatGatewaysOutput, lastPage bool) bool) bool {
-						callback(&ec2.DescribeNatGatewaysOutput{
-							NatGateways: []*ec2.NatGateway{
-								{
-									NatGatewayId: aws.String("nat-0a5408508b19ef490"), // nat1
-								},
-							},
-						}, true)
-						return true
-					})).Return(nil)
+			mocks: func(repo *repository.MockEC2Repository) {
+				repo.On("ListAllNatGateways").Once().Return([]*ec2.NatGateway{
+					{
+						NatGatewayId: aws.String("nat-0bb462a8a4b4bf460"),
```

Comment:
You can reuse the old id and golden file there

Reply:
Right.
Change reverted, you can check again.